File: leetcode_ly/c++/easy/PowerOfTwo.cpp
#include<math.h>

class Solution {
public:
    bool isPowerOfTwo(int n) {
        double result = log10(n) / log10(2);
        return (int) result == result;
    }
};

class Solution {
public:
    bool isPowerOfFour(int num) {
        if(num <= 0){
            return false;
        }
        else{
            while(num != 1){
                int x = num % 4;
                if(x != 0){
                    return false;
                }
                num = num / 4;
            }
            return (num == 1);
        }
    }
};

File: leetcode_ly/java/Bit Manipulation/PowerOfTwo.java
public class Solution {
    public boolean isPowerOfTwo(int n) {
    	//<=0的数不算2的阶乘
        if(n <= 0) return false;
        if(n == 1) return true;
        while(n != 1){
            if(n % 2 == 1) return false;
            n = n / 2;
        }
        return n == 1;
    }
}


//2的阶乘用Bit表示1的个数只有1个，这种方式还不如上面的高效
public class Solution {
    public boolean isPowerOfTwo(int n) {
        if(n <= 0) return false;
        int count = 0;
        while(n != 0){
            n = n & (n - 1);
            count++;
        }
        return count == 1;
    }
}